theseus/core/cost_function.py

```diff
@@ -17,6 +17,19 @@
 from .variable import Variable
 
 
+def _register_vars_in_list(
+    cost_fn: TheseusFunction, var_list_: Iterable[Variable], is_optim: bool = False
```

Comment:
What's our convention for private objects: trailing underscore or leading underscore? We seem to have a mix at the moment so maybe we can just pick one (say leading?).

Reply:
Convention for private is leading underscore. I only use trailing when I have nested function (which this used to be before this PR), to avoid silent scoping errors. I missed the underscore on this, I'll fix. 

If there are other places where I've used trailing to indicate private, that's not intended, and we should fix.

Reply:
Interesting scenario with the nested function. Okay good to know. We'll add in the documentation for conventions. In the final sweep we can find if there are any other trailing underscores.